alf/networks/actor_distribution_networks.py

```diff
@@ -149,11 +199,11 @@ def forward(self, observation, state):
             act_dist (torch.distributions): action distribution
             new_state (nest[tuple]): the updated states
         """
-        encoding = self._encoding_net(observation)
-        encoding, state = self._lstm_encoding_net(encoding, state)
-        act_dist = self._projection_net(encoding)
+        observation, state = Network.forward(self, observation, state)
```

Comment:
Is it possible to wrap this call (```Network.forward```) into a function like ```input_processing(...)``` in base ```Network``` class?  Then this line would be like ```observation, state = self.input_processing(self, observation, state)```, which is more natural and clear in terms what it does. 
The user can also override this ```input_processing``` function in their own class if customization is desirable.

Reply:
I put it in `Network.forward` because that can remind the user of calling it to some extent. Basically it's always beneficial to first call this before you do other things.